crates/sargon/src/profile/v100/profile.rs

```diff
@@ -99,20 +99,28 @@ impl Profile {
         device_factor_source: DeviceFactorSource,
         host_id: HostId,
         host_info: HostInfo,
+        maybe_accounts: Option<impl Into<Accounts>>,
     ) -> Self {
         if !device_factor_source.is_main_bdfs() {
             panic!("DeviceFactorSource is not main BDFS");
         }
         let bdfs = device_factor_source;
         let header =
             Header::new(DeviceInfo::new_from_info(&host_id, &host_info));
+
+        let mainnet_network = match maybe_accounts {
+            None => ProfileNetwork::new_empty_on(NetworkID::Mainnet),
+            Some(accounts) => ProfileNetwork::new_with_accounts(
```

Comment:
Should we not assert that all accounts are controlled by the BDFS? Maybe not, if accounts passed is a mix of Ledger accounts has software.

Hmmm

Reply:
Issue is that this probably seems like a very generic purpose API, while it should be used only when recovering with a Seed Phrase which by definition can be only BDFS currently. It is meant to replace this Host [logic](https://github.com/radixdlt/babylon-wallet-ios/blob/d50309dd03ffcc306c4849d18f28be5b9062403d/RadixWallet/Clients/ProfileStore/ProfileStore.swift#L161).